coordinator/src/bin/coordinator.rs

```diff
@@ -121,13 +127,75 @@ async fn main() -> Result<()> {
+    };
+
+    tokio::spawn({
+        let node = node.clone();
+        async move {
+            loop {
+                let mut conn = node.pool.get().unwrap();
+                match db::positions::Position::get_all_open_positions(&mut conn) {
+                    Ok(positions) => {
+                        let positions = positions
+                            .into_iter()
+                            .filter(|p| {
+                                p.position_state == PositionState::Open
+                                    && OffsetDateTime::now_utc().ge(&p.expiry_timestamp)
+                            })
+                            .collect::<Vec<Position>>();
+
+                        for position in positions.iter() {
+                            tracing::debug!(%position.expiry_timestamp, "Attempting to closed expired position with {}", position.trader);
+                            let channel_id = match node.decide_trade_action(&position.trader) {
+                                Ok(TradeAction::Close(channel_id)) => channel_id,
+                                Ok(_) => {
+                                    tracing::error!(
+                                        ?position,
+                                        "Unable to find sub channel of expired position."
+                                    );
+                                    continue;
+                                }
+                                Err(e) => {
+                                    tracing::error!(
+                                        ?position,
+                                        "Failed to decide trade action. Error: {e:?}"
+                                    );
+                                    continue;
+                                }
+                            };
+
+                            // todo: fetch closing price from bitmex
+                            let closing_price = Decimal::from(30000);
+
+                            match node
+                                .close_position(position, closing_price, channel_id)
+                                .await
+                            {
+                                Ok(_) => tracing::info!(
+                                    "Successfully closed expired position with {}",
+                                    position.trader
+                                ),
+                                Err(e) => tracing::warn!(
+                                    ?position,
+                                    "Failed to close expired position with {}. Error: {e:?}",
+                                    position.trader
+                                ),
+                            }
+                        }
+                    }
+                    Err(e) => {
+                        tracing::error!("Failed to get positions. Error: {e:?}");
+                    }
+                }
+                tokio::time::sleep(Duration::from_secs(600)).await;
```

Comment:
Any particular reason why you went with every 1 minutes?

Reply:
well maybe we need to revisit that. We don't want that interval to be too small as not to interfere with an ongoing closure nor too big to risk not reaching the user when she is online.